tldraw/apps/tldraw-logseq/src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -34,7 +34,15 @@ export const ContextMenu = observer(function ContextMenu({
   }, [])
 
   return (
-    <ReactContextMenu.Root>
+    <ReactContextMenu.Root
+      onOpenChange={open => {
+        if (open && !app.isIn('select.contextMenu')) {
+          app.transition('select').selectedTool.transition('contextMenu')
+        } else if (!open && app.isIn('select.contextMenu')) {
+          app.selectedTool.transition('idle')
+        }
+      }}
+    >
```

Comment:
Although this seems like a nice way to ensure that the app state is correct, it also removes the ability to select an element by right clicking on in, which is why it was removed.

Reply:
Ops, I forgot about this! I am adding back this is to support long-pressing in iOS.